polkadot/xcm/src/lib.rs

```diff
@@ -322,6 +329,25 @@ pub enum VersionedXcm<RuntimeCall> {
 	V5(v5::Xcm<RuntimeCall>),
 }
 
+impl<RuntimeCall> Decode for VersionedXcm<RuntimeCall> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
+		fn decode_variant<T: Decode, I: Input>(input: &mut I) -> Result<T, CodecError> {
+			T::decode_with_depth_limit(MAX_XCM_DECODE_DEPTH, input)
+		}
+
+		Ok(
+			match input.read_byte().map_err(|e| {
+				e.chain("Could not decode `VersionedXcm`, failed to read variant byte")
+			})? {
+				3 => Self::V3(decode_variant(input)?),
+				4 => Self::V4(decode_variant(input)?),
+				5 => Self::V5(decode_variant(input)?),
```

Comment:
I am not a fan of scattering XCM version numbers throughout the codebase - we have constants for that. For example, when we remove `xcm::v3`, the line is removed due to compilation issues rather than keeping the version reference indefinitely.
```suggestion
				v3::VERSION => Self::V3(decode_variant(input)?),
				v4::VERSION => Self::V4(decode_variant(input)?),
				v5::VERSION => Self::V5(decode_variant(input)?),
```

Reply:
Nice. Done